tests/basic/thread.rs

```diff
@@ -35,6 +35,80 @@ fn thread_yield_point() {
     assert_eq!(success_clone.load(Ordering::SeqCst), 0x3);
 }
 
+#[test]
+fn thread_scope() {
+    check_random(
```

Comment:
This should be `check_dfs`. I think the same applies to the two tests below as well.